Description: Parses java.version system property by simply looking at
 the whole number.  This should still work for a >= JDK9 test, since
 java.version strings for JDK8 and earlier will parse to 1 < 9.
Author: tony mancill <tmancill@debian.org>
Forwarded: no

--- a/istack-commons/tools/src/test/java/com/sun/istack/tools/ParallelWorldClassLoaderTest.java
+++ b/istack-commons/tools/src/test/java/com/sun/istack/tools/ParallelWorldClassLoaderTest.java
@@ -142,10 +142,10 @@
     }
 
     private static boolean isJDK9() {
-        String ver = System.getProperty("java.version");
-        int idx = ver.indexOf('-');
+        String ver = System.getProperty("java.specification.version");
+        //int idx = ver.indexOf('-');
         int idx2 = ver.indexOf('.', 1);
-        float v = Float.parseFloat(idx > 0 ? ver.substring(0, idx) : idx2 > 0 ? ver.substring(0, idx2) : ver);
+        float v = Float.parseFloat(idx2 > 0 ? ver.substring(0, idx2) : ver);
         return v > 8;
     }
 }
